crates/pallets/starknet/src/tests/invoke_tx.rs

```diff
@@ -525,3 +529,124 @@ fn test_verify_nonce_in_unsigned_tx() {
+            calldata: vec![
+                Felt252Wrapper::ONE,
+                Felt252Wrapper::from_hex_be(UDC_ADDRESS).unwrap(), // udc address
+                Felt252Wrapper::from_hex_be(UDC_SELECTOR).unwrap(), // deployContract selector
+                Felt252Wrapper::from_hex_be("0x4").unwrap(),       // calldata len
+                transaction_revert_class_hash,                     // contract class hash
+                salt,                                              // salt
+                Felt252Wrapper::ONE,                               // unique
+                Felt252Wrapper::ZERO,                              // constructor calldata len
+            ],
+            max_fee: u128::MAX,
+            offset_version: false,
+        };
+
+        // validate invoke transaction
+        assert_ok!(Starknet::validate_unsigned(
+            TransactionSource::InBlock,
+            &crate::Call::invoke { transaction: deploy_transaction.clone().into() },
+        ));
+
+        let contract_address: FieldElement = get_udc_deployed_contract_address(
+            salt.into(),
+            transaction_revert_class_hash.into(),
+            &UdcUniqueness::Unique(UdcUniqueSettings {
+                deployer_address: FieldElement::from_hex_be(
+                    "0x0642a8b9e2c6cc3a9ddb84575123f262a21415f78db453b0625d889e1e06ac32",
+                )
+                .unwrap(),
+                udc_contract_address: FieldElement::from_hex_be(UDC_ADDRESS).unwrap(),
+            }),
+            &[],
+        );
+
+        // deploy contract
+        assert_ok!(Starknet::invoke(RuntimeOrigin::none(), deploy_transaction.into()));
+
+        let increase_balance_function_selector: &str = "0x362398bec32bc0ebb411203221a35a0301193a96f317ebe5e40be9f60d15320";
+
+        // create increase balance transaction
+        let increase_balance_tx = InvokeTransactionV1 {
+            sender_address: account_addr.into(),
+            signature: vec![],
+            nonce: Felt252Wrapper::TWO,
+            max_fee: u128::MAX,
+            offset_version: false,
+            calldata: vec![
+                Felt252Wrapper::ONE,
+                contract_address.into(),
+                Felt252Wrapper::from_hex_be(increase_balance_function_selector).unwrap(),
+                Felt252Wrapper::from_hex_be("0x1").unwrap(),
+                Felt252Wrapper::from_hex_be("0xa").unwrap(),
+            ],
+        };
+
+        // the transaction reverts and returns Ok
+        assert_ok!(Starknet::invoke(RuntimeOrigin::none(), increase_balance_tx.clone().into()));
+
+        // the storage value should be 0 after the transaction reverts
+        let contract_address = ContractAddress(PatriciaKey(StarkFelt::try_from(contract_address).unwrap()));
+
+        let get_balance_function_selector = EntryPointSelector(
+            StarkFelt::try_from("0x039e11d48192e4333233c7eb19d10ad67c362bb28580c604d67884c85da39695").unwrap(),
```

Comment:
starknet.rs has `get_selector_from_name` which can be used here and at other places where the selector is hardcoded

Reply:
Done